### PR TITLE
Add APIs to fetch pending list and sync_all to graph

### PR DESCRIPTION
**Pull request by tuxdna**
For: https://github.com/openshiftio/openshift.io/issues/1085

Also need:

- [x] Add synced2graph boolean column to versions table - https://github.com/fabric8-analytics/fabric8-analytics-worker/pull/495
- [x] Add PGSQL config to DataModel Importer
- [x]  Invoke /api/v1/pending to enlist all EPVs that need to be synced
- [x]  Invoke /api/v1/sync_all to perform sync to graph

### CODE REVIEW

**Comment by msrb** [comment]
Note we keep data in two separate buckets, first one is `prod-bayesian-core-data` - for EPV data. And the other one is `prod-bayesian-core-package-data` - for EP data, i.e. package-level data (like GitHub for example).

File: src/rest_api.py [hunk 1]
```diff
@@ -46,6 +46,35 @@ def liveness():
...
 
+@app.route('/api/v1/pending')
+def pending():
+    """Generate response for the GET request to /api/v1/pending."""
+    pending_list = data_importer.PostgresHandler().fetch_pending_epvs()
+
+    return flask.jsonify(pending_list), 200
+
+
+@app.route('/api/v1/sync_all')
```
Comment (msrb) [inline src/rest_api.py]: Nice to have feature would be to be able to say that we want to sync all data, but only for example for Maven ecosystem. Maybe something like `/api/v1/sync_all/<ecosystem>`.

But something like that should be pretty easy to add in future.

Comment (tuxdna) [reply src/rest_api.py]: @msrb: Yes that is doable.

Comment (tuxdna) [reply src/rest_api.py]: Latest commit adds ecosystem, package and versions via URL parameters.

File: src/rest_api.py [hunk 2]
```diff
@@ -46,6 +46,49 @@ def liveness():
     return flask.jsonify({}), 200
 
 
+@app.route('/api/v1/pending')
+def pending():
+    """Generate response for the GET request to /api/v1/pending."""
```
Comment (jpopelka) [inline src/rest_api.py]: The comment doesn't say much. What about:
`"""Get request to enlist all the EPVs which are not yet synced to Graph."""`

File: src/rest_api.py [hunk 3]
```diff
@@ -46,6 +46,49 @@ def liveness():
...
+    ecosystem_name = request.args.get('ecosystem', None)
+    package_name = request.args.get('package', None)
+    version_id = request.args.get('version', None)
+
+    pending_list = data_importer.PostgresHandler().fetch_pending_epvs(
+        ecosystem=ecosystem_name,
+        package=package_name,
+        version=version_id)
+
+    input_json = pending_list
```
Comment (jpopelka) [inline src/rest_api.py]: Do we need to use both `input_json` as well as `pending_list` ?
Can we just do directly `input_json = data_importer.PostgresHandler().fetch_pending_epvs(...)` or keep using `pending_list` (that I'd prefer) ?

File: src/rest_api.py [hunk 4]
```diff
@@ -46,6 +46,49 @@ def liveness():
...
+
+    try:
+        report = data_importer.import_epv_from_s3_http(list_epv=input_json)
+        response = {'message': report.get('message'),
+                    'epv': input_json,
+                    'count_imported_EPVs': report.get('count_imported_EPVs')}
+        print(response)
```
Comment (jpopelka) [inline src/rest_api.py]: Looks like left-over debug print.

File: src/rest_api.py (hunk 2 above)
Comment (tuxdna) [reply src/rest_api.py]: Fixed in latest commit.

File: src/rest_api.py (hunk 3 above)
Comment (tuxdna) [reply src/rest_api.py]: Fixed in latest commit.

File: src/rest_api.py (hunk 4 above)
Comment (tuxdna) [reply src/rest_api.py]: Fixed in latest commit.

**Comment by jpopelka** [comment]
I'm fine with the changes, but I think [Michal's comment](https://github.com/fabric8-analytics/fabric8-analytics-data-model/pull/69#issuecomment-357611523) about `prod-bayesian-core-data` vs `prod-bayesian-core-package-data` has still not been resolved, has it ?

**Comment by tuxdna** [comment]
@jpopelka Package level data already gets updated by Data Model Importer

https://github.com/fabric8-analytics/fabric8-analytics-data-model/blob/f79fd8d49f41fc3db949228e09f1064390312e6d/src/data_importer.py#L151

Along with package Version level data

https://github.com/fabric8-analytics/fabric8-analytics-data-model/blob/f79fd8d49f41fc3db949228e09f1064390312e6d/src/data_importer.py#L158

So as long as an EPV gets marked as `synced2graph=FALSE`, it should sync both package and version data.

**Comment by jpopelka** [comment]
I was just pointing to Michal's comment, he should verify whether it's OK. For me it's 👍

**Comment by tuxdna** [comment]
@msrb @jpopelka Thanks for the review!

**Pull request merged**